src/mds/MDSMap.cc

```diff
@@ -1169,3 +1175,28 @@ void MDSMap::set_min_compat_client(ceph_release_t version)
   std::sort(bits.begin(), bits.end());
   required_client_features = feature_bitset_t(bits);
 }
+
+const std::bitset<MAX_MDS>& MDSMap::get_bal_rank_mask_bitset() const {
+  return bal_rank_mask_bitset;
+}
+
+void MDSMap::set_bal_rank_mask(std::string val, std::string bin_string)
+{
+  bal_rank_mask = val;
+  bal_rank_mask_bin_string = bin_string;
+  dout(10) << "set bal_rank_mask_bitset " << bal_rank_mask_bin_string << " with " << bal_rank_mask << dendl;
+}
+
+void MDSMap::update_num_mdss_in_rank_mask_bitset()
+{
+  if (bal_rank_mask_bin_string.length()) {
+    std::bitset<MAX_MDS> _mds_bal_mask_bitset(bal_rank_mask_bin_string);
```

Comment:
```suggestion
    auto _mds_bal_mask_bitset = std::bitset<MAX_MDS>(bal_rank_mask_bin_string);
```

Is more conventional these days.

Reply:
Changed based on your suggestion.